torch/utils/data/datapipes/iter/sharding.py

```diff
@@ -13,15 +13,18 @@
     "ShardingFilterIterDataPipe",
 ]
 
+
 class SHARDING_PRIORITIES(IntEnum):
     DEFAULT = 1
     DISTRIBUTED = 2
     MULTIPROCESSING = 3
 
+
 class _ShardingIterDataPipe(IterDataPipe):
-    def apply_sharding(self, num_of_instances, instance_id, sharding_group):
+    def apply_sharding(self, num_of_instances: int, instance_id: int, sharding_group: SHARDING_PRIORITIES):
```

Comment:
in the changes you've made to the dataloader, you've removed passing `sharding_group` and adjusted the `num_of_instances` and `instance_id` to reflect the process group's `world_size` and `rank`. Are you planning to bring `SHARDING_PRIORITIES` back at some point? If not shouldn't we just remove this enum?

Also FWIW, I realized another point of BC issue is that `num_instances` was changed to `num_of_instances` in `apply_sharding` so if you were calling using kwargs (usually in unittests) it breaks. (e.g. `pipe.apply_sharding(num_instances=4, instance_id=1)` breaks since `num_instances` is now called `num_of_instances`)

Reply:
> Are you planning to bring `SHARDING_PRIORITIES` back at some point? If not shouldn't we just remove this enum?

No, we won't. For `DataLoader2` in TorchData, we will continue to utilize this argument.

> Also FWIW, I realized another point of BC issue is that `num_instances` was changed to `num_of_instances` in `apply_sharding` so if you were calling using kwargs (usually in unittests) it breaks

`DataLoader` is providing positional arguments when calling `pipe.apply_sharding`. Is there any use case that users would call `pipe.apply_sharding(num_instances=4, instance_id=1)` directly?